sdk/core/core-http/lib/util/xml.ts

```diff
@@ -34,4 +78,160 @@ export function parseXML(str: string): Promise<any> {
       });
     }
   });
+  return result;
+}
+
+/**
+ * The XML body to convert to JSON
+ * @param body
+ */
+export async function convertAtomXmlToJson(body: string): Promise<any> {
+  const parser = new xml2js.Parser(getDefaultSettingsForAtomXmlOperations());
+  const result = await new Promise((resolve, reject) => {
+    parser.parseString(removeBOM(body.toString()), function(err: any, parsedBody: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(parsedBody);
+      }
+    });
+  });
+  return result;
+}
+
+/**
+ * @param content The content as it is to be serialized. It should include any root node(s).
+ */
+export function convertJsonToAtomXml(content: any): string {
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+  let doc = xmlbuilder.create();
+
+  doc = doc
+    .begin("entry", { version: "1.0", encoding: "utf-8", standalone: "yes" })
+    .att("xmlns", "http://www.w3.org/2005/Atom");
+
+  doc = doc.ele("updated", new Date().toISOString()).up();
+
+  doc = writeElementValue(doc, "content", content);
+  return doc.doc().toString();
+}
+
+/**
+ * @ignore
+ * Gets the default xml2js settings applicable for Atom based XML operations.
+ */
+function getDefaultSettingsForAtomXmlOperations(): any {
+  const xml2jsSettings = Object.assign({}, xml2jsDefaults);
```

Comment:
As discussed with @chradek offline, we should avoid creating the json object from scratch for every incoming request. This can be avoided by having 2 separate constants, one for normal xml and the other for atom xml.

With these 2 constants defined, we have no need for either `getDefaultSettings()` or `getDefaultSettingsForAtomXmlOperations()`, as the constants can be used directly

Reply:
Agree, updated